packages/core/schematics/ng-generate/control-flow-migration/migration.ts

```diff
@@ -68,10 +68,12 @@ export function migrateTemplate(
     // whether it's safe to remove the CommonModule to the
     // original component class source file
     if (templateType === 'templateUrl' && analyzedFiles !== null &&
-        analyzedFiles.has(file.sourceFilePath)) {
-      const componentFile = analyzedFiles.get(file.sourceFilePath)!;
+        analyzedFiles.has(file.sourceFile.fileName)) {
+      const componentFile = analyzedFiles.get(file.sourceFile.fileName)!;
+      componentFile.getSortedRanges();
       componentFile.removeCommonModule = file.removeCommonModule;
       componentFile.canRemoveImports = file.canRemoveImports;
+      componentFile.verifyCanRemoveImports();
```

Comment:
nit: it might be a bit better to compute the value in a separate function and assign the value like this:
```suggestion
      componentFile.removeCommonModule = canRemoveImports(file);
      componentFile.canRemoveImports = file.canRemoveImports;
```
Currently we are changing the value of the `componentFile.removeCommonModule` within the `verifyCanRemoveImports` function, which is a bit unexpected.

Reply:
I get that this is confusing. So I'll add a comment here to clarify what's happening, but the TL;DR is that the `file.canRemoveImports` and `file.removeCommonModule` come from the templateUrl `html` file. `verifyCanRemoveImports` is running on the associated `.ts` file with the component class. We determine if the template is ok with removal of common module and associated imports, which is passed in to the associated component file, and then there's a verification step against the symbols afterwards just in case it's still unsafe. So two verifications are happening here.